### PR TITLE
update package name

### PR DESCRIPTION
**Pull request by puneet-pdx**
- Update package name to `com.esri.arcgisruntime.opensourceapps.mapsapp`
- update RELEASE.md
adds entry for new release, also fixes release 1.0.4 entry
https://github.com/Esri/maps-app-android/releases

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
-# Release 1.0.4
+# Release 1.0.6
```
Comment (changanxian) [inline RELEASE.md]: The version number is ahead of the other two `1.0.6`, I assume it is correct

Comment (eri9600) [reply RELEASE.md]: I am also a bit confused because this one is 1.0.6 and the others are 1.0.5

Comment (puneet-pdx) [reply RELEASE.md]: @changanxian @eri9600  yes, it is correct. `maps-app-android` has had an additional patch release where its minimum API level was updated to 21. The other two apps were already at that level.

Comment (puneet-pdx) [reply RELEASE.md]: https://github.com/Esri/maps-app-android/releases
 
https://github.com/Esri/mapbook-android/releases

https://github.com/Esri/nearby-android/releases

Comment (changanxian) [reply RELEASE.md]: ok

**Pull request merged**